### PR TITLE
fix: update refresh label logic

### PR DESCRIPTION
**Pull request by sofisl**
@bcoe and I applied the following fixes after testing the bot:

1. lint
2. Made sure the bot checked for mismatching descriptions on preexisting labels, and made sure the descriptions were short enough so that the bot would update the descriptions on the labels
3. Made sure the bot didn't try deleting labels if there were no labels (i.e., new repo)
4. When a new repo is created, the bot creates labels and updates the ones that come prepackaged w/i github. This means that the bot triggers itself because it is also triggered when a label is edited. This can cause issues because it tries to create the same label more than once. We've updated the logic so that it doesn't log errors with the code 'already_created.'

### CODE REVIEW

File: packages/label-sync/src/label-sync.ts [hunk 1]
```diff
@@ -73,12 +77,14 @@ export = (app: Application) => {
     const { owner, repo } = context.repo();
     // TODO: Limit this to pushes that edit `labels.json`
     if (
-      owner === 'googleapis' &&
+      owner === 'sofisl' &&
```
Comment (bcoe) [inline packages/label-sync/src/label-sync.ts]: looks like you left in some testing data here, this should be `googleapis`.

File: packages/label-sync/src/label-sync.ts [hunk 2]
```diff
@@ -41,19 +42,22 @@ interface Repos {
...
 }
 
-async function refreshLabels() {
-  const url =
-    'https://github.com/googleapis/repo-automation-bots/blob/master/packages/label-sync/src/labels.json';
-  const res = await request<Labels>({ url });
-  labelsCache = res.data;
+async function refreshLabels(github: GitHubAPI) {
+  const data = (await github.repos.getContents({
+    owner: 'sofisl',
```
Comment (bcoe) [inline packages/label-sync/src/label-sync.ts]: this should be `googleapis` too.

File: packages/label-sync/src/label-sync.ts [hunk 3]
```diff
@@ -73,12 +77,14 @@ export = (app: Application) => {
...
     if (
-      owner === 'googleapis' &&
+      owner === 'sofisl' &&
       repo === 'repo-automation-bots' &&
       context.payload.ref === 'refs/heads/master'
     ) {
-      await refreshLabels();
+      await refreshLabels(context.github);
       // TODO: Use the GitHub installations API to retreive this list
+      //   (also, our QA indicates there's a good chance this request isn't
```
Comment (bcoe) [inline packages/label-sync/src/label-sync.ts]: I think we can get rid of thee comments, seems like your QA showed us things were working.

File: packages/label-sync/src/label-sync.ts [hunk 4]
```diff
@@ -156,23 +163,24 @@ async function reconcileLabels(github: GitHubAPI, owner: string, repo: string) {
...
-          console.log(`Deleted '${l.name}' from ${owner}/${repo}`);
-        })
-        .catch(e => {
-          console.error(`Error deleting label ${l.name} in ${owner}/${repo}`);
-          console.error(e.stack);
-        });
-      promises.push(p);
+  if (!Object.keys(oldLabels).length) {
+    for (const l of oldLabels) {
+      console.log(l.name);
```
Comment (bcoe) [inline packages/label-sync/src/label-sync.ts]: let's get rid of this label, but perhaps add something more descriptive right above `await github.issues`.

```
console.info(`deleting label ${l.name}`)
```

File: packages/label-sync/package.json [hunk 5]
```diff
@@ -25,9 +25,11 @@
   },
   "dependencies": {
     "@octokit/rest": "^16.34.0",
+    "@types/uuid": "^3.4.6",
     "gaxios": "^2.1.0",
     "gcf-utils": "1.1.0",
-    "probot": "9.6.4"
+    "probot": "9.6.4",
+    "uuid": "^3.3.3"
```
Comment (bcoe) [inline packages/label-sync/package.json]: we didn't end up using `uuid`, so let's get rid of this.

Comment (bcoe) [reply packages/label-sync/package.json]:
```
npm uninstall uuid
npm uninstall @types/uuid
```

File: packages/label-sync/src/labels.json [hunk 6]
```diff
@@ -62,7 +62,7 @@
     },
     {
       "name": "priority: p0",
-      "description": "Highest priority. Critical issue. P0 implies highest priority (e.g. blocking imminent release). P0 issues should be monitored on a daily basis.",
```
Comment (bcoe) [inline packages/label-sync/src/labels.json]: GitHub had a limit on the length of label description @JustinBeckwith; awesome debugging @sofisl.

I believe this was the error that we were seeing when we tried to create labels in parallel with `Promise.all`, I still think for the time being this approach is reasonable however, it's helped us debug a couple issues more easily to run in series.

File: packages/label-sync/src/label-sync.ts (hunk 2 above)
Comment (sofisl) [reply packages/label-sync/src/label-sync.ts]: great catch!

File: packages/label-sync/src/label-sync.ts (hunk 4 above)
Comment (sofisl) [reply packages/label-sync/src/label-sync.ts]: Just deleted line 165, as it logs at the end if the label was deleted or not!

File: packages/label-sync/src/label-sync.ts [hunk 7]
```diff
@@ -17,6 +17,7 @@
 import { Application } from 'probot';
 import { request } from 'gaxios';
 import { GitHubAPI } from 'probot/lib/github';
+import e = require('express');
```
Comment (bcoe) [inline packages/label-sync/src/label-sync.ts]: I don't think we need this import.

File: packages/label-sync/src/label-sync.ts [hunk 8]
```diff
@@ -140,12 +142,14 @@ async function reconcileLabels(github: GitHubAPI, owner: string, repo: string) {
           name: l.name,
         })
         .catch(e => {
-          console.error(`Error creating label ${l.name} in ${owner}/${repo}`);
-          console.error(e.stack);
+          //ignores errors that are caused by two requests kicking off at the same time
+          if (e.errors[0].code !== 'already_exists') {
```
Comment (bcoe) [inline packages/label-sync/src/label-sync.ts]: love this 😄

**Comment by bcoe** [comment]
@sofisl thanks for thoroughly testing this :tada:

**Pull request merged**